assets/js/plugin-check-admin.js

```diff
@@ -207,4 +213,85 @@
 
 		return data;
 	}
+
+	/**
+	 * Renders results for each check on the page.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} results The results object.
+	 */
+	function renderResults( results ) {
+		const { errors, warnings } = results;
+
+		// Render errors and warnings for files.
+		for ( const file in errors ) {
+			if ( warnings[ file ] ) {
+				renderFileResults( file, errors[ file ], warnings[ file ] );
+				delete warnings[ file ];
+			} else {
+				renderFileResults( file, errors[ file ], [] );
+			}
+		}
+
+		// Render remaining files with only warnings.
+		for ( const file in warnings ) {
+			renderFileResults( file, [], warnings[ file ] );
+		}
+	}
+
+	/**
+	 * Renders the file results table.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} file     The file name for the results.
+	 * @param {Object} errors   The file errors.
+	 * @param {Object} warnings The file warnings.
+	 */
+	function renderFileResults( file, errors, warnings ) {
+		const tableTemplate = wp.template( 'plugin-check-results-table' );
+		const index = Date.now();
+
+		// Render the file table.
+		resultsContainer.innerHTML += tableTemplate( { file, index } );
+		const resultsTable = document.getElementById(
+			'plugin-check__results-body-' + index
+		);
+
+		// Render results to the table.
+		renderResultRows( 'ERROR', errors, resultsTable );
+		renderResultRows( 'WARNING', warnings, resultsTable );
+	}
+
+	/**
+	 * Renders a result row onto the file table.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {string} type    The result type. Either ERROR or WARNING.
+	 * @param {Object} results The results object.
+	 * @param {Object} table   The HTML table to append a result row to.
+	 */
+	function renderResultRows( type, results, table ) {
+		const rowTemplate = wp.template( 'plugin-check-results-row' );
```

Comment:
Same here, since this function is also called several times.

I think to fix that in both instance, we should introduce a general function that handles the template rendering generically and is smart about it, something like:
```js
const templates = {};
function renderTemplate( templateSlug, data ) {
	if ( ! templates[ templateSlug ] ) {
		templates[ templateSlug ] = wp.template( templateSlug );
	}
	const template = templates[ templateSlug ];
	return template( data );
}
```